File: packages/image-labeling/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/imagelabeling/classes/options/ProcessImageOptions.java
package io.capawesome.capacitorjs.plugins.mlkit.imagelabeling.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.mlkit.imagelabeling.ImageLabelingPlugin;

public class ProcessImageOptions {

    @NonNull
    private final String path;

    private final float confidenceThreshold;

    public ProcessImageOptions(@NonNull PluginCall call) throws Exception {
        String path = call.getString("path", null);
        if (path == null) {
            throw new Exception(ImageLabelingPlugin.ERROR_PATH_MISSING);
        }
        this.path = path;
        this.confidenceThreshold = call.getFloat("confidenceThreshold", 0.5f);
    }

    @NonNull
    public String getPath() {
        return path;
    }

    public float getConfidenceThreshold() {
        return confidenceThreshold;
    }
}

File: packages/image-labeling/ios/Plugin/Classes/Options/ProcessImageOptions.swift
import Foundation
import Capacitor

@objc public class ProcessImageOptions: NSObject {
    private var path: String
    private var confidenceThreshold: Float

    init(_ call: CAPPluginCall) throws {
        guard let path = call.getString("path") else {
            throw CustomError.pathMissing
        }
        self.path = path
        self.confidenceThreshold = call.getFloat("confidenceThreshold", 0.5)
    }

    func getPath() -> String {
        return path
    }

    func getConfidenceThreshold() -> Float {
        return confidenceThreshold
    }
}
